File: lib/models/service.dart
class Service {
  final String id;
  final String name;
  final double price;
  final String description;
  final String providerId;

  Service({
    required this.id,
    required this.name,
    required this.price,
    required this.description,
    required this.providerId,
  });

  factory Service.fromJson(Map<String, dynamic> json) {
    return Service(
      id: json['_id'],
      name: json['name'],
      price: (json['price'] as num).toDouble(),
      description: json['description'],
      providerId: json['provider'],
    );
  }
}

File: server/models/service.js
const mongoose=require('mongoose');

const serviceSchema=new mongoose.Schema({
  name:{
    type:String,
    required:true
  },
  price:{
    type:Number,
    required:true,
  },
  provider:{
    type:mongoose.Schema.Types.ObjectId,
    ref:'Provider',
    required:true
  },
  description:{
    type:String,
    required:true
  }
});

const Service=mongoose.model('Service',serviceSchema);



module.exports=Service;